backend/internal/ou/composite_store.go

```diff
@@ -0,0 +1,268 @@
+	return c.dbStore.UpdateOrganizationUnit(ou)
+}
+
+// DeleteOrganizationUnit deletes an organization unit from the database store only.
+// Immutability and children validation are handled at the service layer.
+func (c *compositeOUStore) DeleteOrganizationUnit(id string) error {
+	return c.dbStore.DeleteOrganizationUnit(id)
+}
+
+// CheckOrganizationUnitHasChildResources checks if an OU has child resources in either store.
+func (c *compositeOUStore) CheckOrganizationUnitHasChildResources(id string) (bool, error) {
+	return immutableresource.CompositeHasChildrenHelper(
+		func() (bool, error) { return c.dbStore.CheckOrganizationUnitHasChildResources(id) },
+		func() (bool, error) { return c.fileStore.CheckOrganizationUnitHasChildResources(id) },
+	)
+}
+
+// GetOrganizationUnitChildrenCount retrieves the count of child OUs from both stores.
+func (c *compositeOUStore) GetOrganizationUnitChildrenCount(id string) (int, error) {
+	return immutableresource.CompositeMergeCountHelper(
+		func() (int, error) { return c.dbStore.GetOrganizationUnitChildrenCount(id) },
+		func() (int, error) { return c.fileStore.GetOrganizationUnitChildrenCount(id) },
+	)
+}
+
+// GetOrganizationUnitChildrenList retrieves child OUs from both stores with pagination.
+func (c *compositeOUStore) GetOrganizationUnitChildrenList(
+	id string, limit, offset int) ([]OrganizationUnitBasic, error) {
+	// Get counts from both stores first to determine fetch strategy
+	dbCount, err := c.dbStore.GetOrganizationUnitChildrenCount(id)
+	if err != nil {
+		return nil, err
+	}
+
+	fileCount, err := c.fileStore.GetOrganizationUnitChildrenCount(id)
+	if err != nil {
+		return nil, err
+	}
+
+	totalCount := dbCount + fileCount
+
+	// If offset is beyond total count, return empty
+	if offset >= totalCount {
+		return []OrganizationUnitBasic{}, nil
+	}
+
+	// Fetch all children from both stores to ensure correct pagination
+	// This is safe because OUs typically have a reasonable number of children
+	dbChildren, err := c.dbStore.GetOrganizationUnitChildrenList(id, dbCount, 0)
+	if err != nil {
+		return nil, err
+	}
+
+	fileChildren, err := c.fileStore.GetOrganizationUnitChildrenList(id, fileCount, 0)
+	if err != nil {
+		return nil, err
+	}
+
+	// Merge results and deduplicate by ID (defensive programming)
+	allChildren := mergeAndDeduplicateChildren(dbChildren, fileChildren)
+
+	// Apply pagination to merged results
+	if offset >= len(allChildren) {
+		return []OrganizationUnitBasic{}, nil
+	}
+
+	end := offset + limit
+	if end > len(allChildren) {
+		end = len(allChildren)
+	}
+
+	return allChildren[offset:end], nil
+}
+
+// mergeAndDeduplicateOUs merges root-level OUs from both stores and removes duplicates by ID.
+// While duplicates shouldn't exist by design (an OU exists in only one store), this provides
+// defensive programming against misconfigurations or bugs.
+func mergeAndDeduplicateOUs(dbOUs, fileOUs []OrganizationUnitBasic) []OrganizationUnitBasic {
```

Comment:
shall we move the private function to the bottom of the file and bring the interface methods up.

Reply:
Fixed